Sources/SwiftKafka/RDKafka/RDKafkaConfig.swift

```diff
@@ -0,0 +1,103 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+
+/// A collection of helper functions wrapping common `rd_kafka_conf_*` functions in Swift.
+struct RDKafkaConfig {
```

Comment:
Can we make this a class that holds the `rd_kafka_conf_new` pointer in an optional. If the optional is `some` we are going to destroy it in `deinit`. Additionally we offer a method that creates a new `rd_kafka` and sets the optional to `nil`. This way we can ensure in a single class that nothing breaks this.

Reply:
I think our Package should **not** keep a reference to the `OpaquePointer` of the `rd_kafka_conf_t`. This is because in `KafkaClient`, we invoke [`rd_kafka_new`](https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#a63d5cd86ab1f77772b2be170e1c09c24) which takes the `rd_kafka_conf_t` pointer we just created as an argument and assumes ownership of it (meaning it frees it once the new Kafka connection has been established). `RDKafkaConfig.createFrom` should therefore just serve as a wrapper for `rdkafka` C functions and not contain any more pointer ownership logic as this is handled by `rdkafka` itself already.